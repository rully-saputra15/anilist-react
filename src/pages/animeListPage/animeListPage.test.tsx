import { expect, it, vi, describe, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";
import AnimeListPage from "./animeListPage";
import ReactDOM from "react-dom";

describe("AnimeListPage", () => {
  beforeEach(() => {
    render(
      <AnimeListPage
        isLoading={false}
        isBulkMode={false}
        currentPage={0}
        selectedAnime={[]}
        observerTarget={{ current: null }}
        handleGoToAnimeDetail={vi.fn()}
        handlePreviousPage={vi.fn()}
        handleNextPage={vi.fn()}
        handleEnableBulkMode={vi.fn()}
        handleAddSelectedAnime={vi.fn()}
        handleConfirmBulkAdd={vi.fn()}
      />
    );
  });
  it("renders without crashing when bulk mode", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <AnimeListPage
        isLoading={false}
        isBulkMode={true}
        currentPage={0}
        selectedAnime={[]}
        observerTarget={{ current: null }}
        handleGoToAnimeDetail={vi.fn()}
        handlePreviousPage={vi.fn()}
        handleNextPage={vi.fn()}
        handleEnableBulkMode={vi.fn()}
        handleAddSelectedAnime={vi.fn()}
        handleConfirmBulkAdd={vi.fn()}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders without crashing when not bulk mode", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <AnimeListPage
        isLoading={false}
        isBulkMode={false}
        currentPage={0}
        selectedAnime={[]}
        observerTarget={{ current: null }}
        handleGoToAnimeDetail={vi.fn()}
        handlePreviousPage={vi.fn()}
        handleNextPage={vi.fn()}
        handleEnableBulkMode={vi.fn()}
        handleAddSelectedAnime={vi.fn()}
        handleConfirmBulkAdd={vi.fn()}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders AnimePageList correctly", () => {
    // const tree = render(
    //   <AnimeListPage
    //     isLoading={false}
    //     isBulkMode={false}
    //     currentPage={0}
    //     selectedAnime={[]}
    //     observerTarget={{ current: null }}
    //     handleGoToAnimeDetail={vi.fn()}
    //     handlePreviousPage={vi.fn()}
    //     handleNextPage={vi.fn()}
    //     handleEnableBulkMode={vi.fn()}
    //     handleAddSelectedAnime={vi.fn()}
    //     handleConfirmBulkAdd={vi.fn()}
    //   />
    // );
    // expect(tree).toMatchSnapshot(``);
    // console.log(screen.getByText(/Bulk Add Collection/i));
    // expect(screen.getByText(/Bulk Add Collection/i)).toBeInTheDocument();
  });
});
