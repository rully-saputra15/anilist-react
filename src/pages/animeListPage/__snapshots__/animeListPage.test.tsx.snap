// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AnimeListPage > renders AnimePageList correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="css-11t7uif-AnimeListPage"
        >
          <div
            class="css-t37or7-AnimeListPage"
          >
            <button
              class="css-go84mi-ButtonBasic"
            >
              &lt;
            </button>
            <span>
              0
            </span>
            <button
              class="css-go84mi-ButtonBasic"
            >
              &gt;
            </button>
          </div>
          <button
            class="css-go84mi-ButtonBasic"
          >
            Bulk Add Collection
          </button>
        </div>
        <div
          class="css-1jma2nr-AnimeListPage"
        />
        <div
          class="css-118qxnh-AnimeListPage"
        >
          Load More
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="css-11t7uif-AnimeListPage"
      >
        <div
          class="css-t37or7-AnimeListPage"
        >
          <button
            class="css-go84mi-ButtonBasic"
          >
            &lt;
          </button>
          <span>
            0
          </span>
          <button
            class="css-go84mi-ButtonBasic"
          >
            &gt;
          </button>
        </div>
        <button
          class="css-go84mi-ButtonBasic"
        >
          Bulk Add Collection
        </button>
      </div>
      <div
        class="css-1jma2nr-AnimeListPage"
      />
      <div
        class="css-118qxnh-AnimeListPage"
      >
        Load More
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
